sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1829,6 +1830,64 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Return scholastic standing application
+   * @param locationId location id.
+   * @param applicationId application id.
+   * @returns Application.
+   */
+  async processScholasticStanding(
+    locationId: number,
+    applicationId: number,
+  ): Promise<Application> {
+    const application = await this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application",
+        "currentAssessment.id",
+        "offering.id",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "student.id",
+        "user.id",
+        "user.firstName",
+        "user.lastName",
+        "user.email",
+        "studentAppeal.id",
+      ])
+      .innerJoin("application.currentAssessment", "currentAssessment")
+      .innerJoin("currentAssessment.offering", "offering")
+      .leftJoin("currentAssessment.studentAppeal", "studentAppeal")
+      .innerJoin("application.location", "location")
+      .innerJoin("application.student", "student")
+      .innerJoin("student.user", "user")
+      .where("application.id = :applicationId", { applicationId })
+      .andWhere("location.id = :locationId", { locationId })
+      .getOne();
+
+    if (!application) {
+      throw new CustomNamedError(
+        "Application Not found or invalid current assessment or offering.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (application.isArchived) {
+      throw new CustomNamedError(
+        "This application is no longer eligible to request changes.",
+        APPLICATION_CHANGE_NOT_ELIGIBLE,
+      );
+    }
+
+    if (application.applicationStatus !== ApplicationStatus.Completed) {
+      throw new CustomNamedError(
+        "Cannot report a change for application with status other than completed.",
+        INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+      );
+    }
```

Comment:
These validations can be done at the controller itself, these custom errors are processed back by controller.